upf/interop/to_tarski.py

```
@@ -0,0 +1,233 @@
+        assert len(args) == 1
+        variables = [self.lang.variable(v.name(), self.lang.get_sort(v.type().name())) for v in expression.variables()] # type: ignore
+        return tarski.syntax.forall(*variables, args[0])
+
+    def walk_equals(self, expression: 'upf.model.FNode', args: List['tarski.syntax.formulas.Formula']) -> 'tarski.syntax.formulas.Formula':
+        assert len(args) == 2
+        return tarski.syntax.Atom(self.lang.get_predicate(tarski.syntax.BuiltinPredicateSymbol.EQ), args)
+
+    def walk_le(self, expression: 'upf.model.FNode', args: List['tarski.syntax.formulas.Formula']) -> 'tarski.syntax.formulas.Formula':
+        assert len(args) == 2
+        return tarski.syntax.Atom(self.lang.get_predicate(tarski.syntax.BuiltinPredicateSymbol.LE), args)
+
+    def walk_lt(self, expression: 'upf.model.FNode', args: List['tarski.syntax.formulas.Formula']) -> 'tarski.syntax.formulas.Formula':
+        assert len(args) == 2
+        return tarski.syntax.Atom(self.lang.get_predicate(tarski.syntax.BuiltinPredicateSymbol.LT), args)
+
+    def walk_fluent_exp(self, expression: 'upf.model.FNode', args: List['tarski.syntax.formulas.Formula']) -> 'tarski.syntax.formulas.Formula':
+        tarski_fluent_rep = self.lang.get(expression.fluent().name())
+        new_args = [x.constant_value() if x.is_int_constant() else args[i] for i, x in enumerate(expression.args())]
+        new_args = []
+        for i, x in enumerate(expression.args()):
+            if x.is_int_constant():
+                type = (expression.fluent().signature())[i]
+                typename = _type_name_added_to_language_if_needed(self.lang, type)
+                constant = tarski.syntax.Constant(x.int_constant_value(), \
+                    self.lang.get_sort(typename))
+                new_args.append(constant)
+            elif x.is_real_constant():
+                pass
```

Here we can raise an exception